src/sentry/static/sentry/app/components/clipboard.tsx

```diff
@@ -1,91 +1,64 @@
 import React from 'react';
-import ReactDOM from 'react-dom';
-import Clip from 'clipboard';
+import copy from 'copy-text-to-clipboard';
 
 import {addErrorMessage, addSuccessMessage} from 'app/actionCreators/indicator';
 
-type DefaultProps = {
-  successMessage: string;
-  errorMessage: string;
-  hideMessages: boolean;
-};
-
 type Props = {
+  children: React.ReactElement<{onClick: () => void}>;
   value: string;
+  successMessage?: string;
+  errorMessage?: string;
+  hideMessages?: boolean;
   hideUnsupported?: boolean;
   onSuccess?: () => void;
   onError?: () => void;
-} & DefaultProps;
+};
 
-class Clipboard extends React.Component<Props> {
-  static defaultProps: DefaultProps = {
-    hideMessages: false,
-    successMessage: 'Copied to clipboard',
-    errorMessage: 'Error copying to clipboard',
-  };
+/**
+ * copy-text-to-clipboard relies on `document.execCommand('copy')`
+ */
+function isSupported() {
+  const support = !!document.queryCommandSupported;
+  return support && !!document.queryCommandSupported('copy');
+}
 
-  componentWillUnmount() {
-    if (this.clipboard) {
-      this.clipboard.destroy();
+function Clipboard({
+  children,
+  value,
+  successMessage = 'Copied to clipboard',
+  errorMessage = 'Error copying to clipboard',
+  hideMessages = false,
+  hideUnsupported,
+  onSuccess,
+  onError,
+}: Props) {
+  function handleClick() {
+    // Copy returns whether it succeeded to copy the text
+    const success = copy(value);
+    if (success) {
+      if (!hideMessages) {
+        addSuccessMessage(successMessage);
+      }
+      onSuccess?.();
+    } else {
+      if (!hideMessages) {
+        addErrorMessage(errorMessage);
+      }
+      onError?.();
```

Comment:
can we maybe remove the `else` ?

```suggestion
    if (success) {
      if (!hideMessages) {
        addSuccessMessage(successMessage);
      }
      onSuccess?.();
      return;
    } 
    
      if (!hideMessages) {
        addErrorMessage(errorMessage);
      }
      onError?.();
```